### DISCUSSION TITLE
how to check whether database supports column type

### DISCUSSION THREAD

**Discussion by HoneyryderChuck** [jeremyevans/sequel#2221] [General]
I'm expanding the test suite of one of my gems to support sqlserver, via the tinytds adapter. I've seen that my tests rely on on the database [supporting a JSON type, which tinyds adapter does not support](https://gitlab.com/os85/tobox/-/jobs/7897445084) (although [recent versions of sqlserver seem to support it?](https://learn.microsoft.com/en-us/sql/relational-databases/json/json-data-sql-server?view=sql-server-ver16)). 

In order to fallback to `TEXT` column in such cases, I'd need a way to verify support for a given datatype in the migration:

```ruby
# smth like
if supports_datatype?(:json)
  column :data_before, :json, null: true
else
 column :data_before, :text, null: true
```

but I didn't find a function I could use for that effect. Is there one?

**Comment by jeremyevans** [comment]
There currently isn't a database-independent method to use to check whether a type is supported.  I'm guessing the only generic approach that would work would be attempting to create a single-column table with the type, and see if it fails, and removing the table afterward.  Even that is not 100%, because there are other reasons table creation could fail, such as lack of permissions.
